Projects/soupstone.md

```diff
@@ -3,19 +3,95 @@
 * [On Github](https://github.com/jamesmunns/soupstone)
 
 ## Summary
-A fun little electronic fidget toy (eventually) with the help of [Embassy](https://github.com/embassy-rs/embassy)
 
-## Goals
+A fun little electronic fidget toy (eventually). The goal is to make a small, battery powered device that has a bunch of fun inputs and outputs, like clicky switches and a piezo speaker, which responds to your inputs.
+
+Ideally: It should be very easy to hack on, and write fun random-ish behaviors, to make it more interesting to play with.
+
+I am currently using the [Seeed Studio XIAO nRF52840 Sense] as the main MCU/SOM, and [Embassy] for software and driver support.
+
+The name comes from the [Stone Soup] folk story, as well as [worry stones].
 
+[Embassy]: https://github.com/embassy-rs/embassy
+[Seeed Studio XIAO nRF52840 Sense]: https://wiki.seeedstudio.com/XIAO_BLE/
+[Stone Soup]: https://en.wikipedia.org/wiki/Stone_Soup
+[worry stone]: https://en.wikipedia.org/wiki/Worry_stone
 
+## Goals
+
+* Make it quick and easy to iterate on
+* Make it robust enough to carry around and use continuously
+* Don't make it too complicated, it's a toy!
+* Build a few units to share with friends and family, and as a neat gift
+* Maybe make it something nice to use later for teaching embedded Rust
 
 ## Current State
 
+As of March 2023, still pretty early! I've been starting with the [nRF52840-DK], because I have some laying around, and they have both a debugger and the MCU USB device port broken out.
+
+The initial work has been focused on writing a very simple "[stage zero]" USB RAM loader (see below). As the XIAO board does not have easily-accessible SWD debugging pins for reprogramming, it will generally be easier to reprogram and debug the device via the main USB port.
+
+I have quite a few parts like toggle/momentary switches, analog potentiometers, rotary encoders, and LEDs in my parts drawers. The next step will be building some basic proto boards to decide what is fun, move them to PCBs, and make some initial rough units to play with.
 
+[nRF52840-DK]: https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dk
+[stage zero]: https://github.com/jamesmunns/soupstone/pull/2
 
 ## Areas of research
 
+### No Debugger
 
-## Note logs
+In Rust, it is very common to use an external debugging probe via SWD, like a [J-Link] or [DAPLink]. This makes it really easy to reprogram the device, and even get logs via something like [probe-run] and [defmt]. These are made possible using [probe-rs]'s
```

Comment:
This section seems to be nicely elaborated, it is just a concept I am unfamiliar with (and would have to read a few times over / read up more in general)

Reply:
ah i see this is "outdated" but I just checked the updated one. I still stick with my original comment though haha

Reply:
Yeah, this is a topic that makes a bit more sense if you are familiar with embedded development, but I don't know if I want to entirely expand that here.

I'll open an issue for explaining how debugging probes work.